Add delete links for images

lzhang0616 opened the pull request.

bboe commented on app/views/layouts/application.html.erb:
```diff
@@ -7,11 +7,14 @@
 <body>
-  <% if flash[:error].present? %>
-    <p>
+  <p>
```
Do we want this `p` tag to appear when there are no flash messages?

lzhang0616 replied:
I have fixed this, so that the paragraph tag only shows when there is a flash message. Please see the commit [484c3e9](https://github.com/AppFolioOnboarding/Li_Image_Share/commit/484c3e9c4b68f4ebb5db7fe16cfa6b16163366d4)

bboe commented on test/controllers/images_controller_test.rb:
```diff
@@ -2,44 +2,73 @@
 
-  test "display all saved images on index page" do
+  test 'fail to show an image' do
+    Image.create!(url: VALID_IMAGE_URL)
```
Do we need to create an image for this test?

bboe commented on test/controllers/images_controller_test.rb:
```diff
@@ -2,44 +2,73 @@
+    Image.create!(url: VALID_IMAGE_URL)
+
+    assert_raises ActiveRecord::RecordNotFound do
+      get :show, id: Image.last.id + 1
```
If tests were to run in parallel on the same database what could go wrong?

lzhang0616 replied:
If the creation happens first, then this code is fine. However, if the query of Image.last happens first, then the exception will raise since there is no record in database yet. In this case, the test will past by default. I think as Macklin suggested below, I can just use -1 as the invalid ID here.

bboe commented on test/controllers/images_controller_test.rb:
```diff
@@ -2,44 +2,73 @@
+  test 'delete a saved image with a valid id' do
+    Image.create!(url: VALID_IMAGE_URL)
+
+    delete :destroy, id: Image.first.id
```
Should we delete the one we just created? If there are others already then perhaps the `create!` isn't necessary.

lzhang0616 replied:
@bboe I have avoid this unnecessary create! in the "use -1 as invalid id and add local variable image to avoid using Image.first" commit

bboe commented on test/controllers/images_controller_test.rb:
```diff
@@ -2,44 +2,73 @@
+  end
+
+  test 'fail to delete a saved image with an invalid id' do
+    Image.create!(url: VALID_IMAGE_URL)
```
Is this `create!` necessary?

lzhang0616 replied:
I have avoid this unnecessary create! in the "use -1 as invalid id and add local variable image to avoid using Image.first" commit ([959f470](https://github.com/AppFolioOnboarding/Li_Image_Share/commit/959f4701e583beff77687f2a1ed47b9280044253))

lzhang0616 commented:
We rebased and squashed some commits to make the history cleaner. You can review the old commit comments here: https://github.com/AppFolioOnboarding/Li_Image_Share/compare/3450ffab1f9a17fdf6593b392ab38019e4def7be...2aadbeb8d0b61c5169d6a9de8852eaa77343ad56
